### PR TITLE
fix: send empty string instead of null when the schema data is undefined

### PR DESCRIPTION
**Pull request by sandeepdsvs**
## What are the changes introduced in this PR?

This PR is just for reviewing the changes made to fix the following issue discussed in [this thread](https://rudderlabs.slack.com/archives/C02A3UAU9UN/p1702994540732409) for Zen Business customer. This PR is just for review and should not be merged as this branch is used for dedicated deployment for transformer.

## Please explain the objectives of your changes below

This PR will make sure empty string, `''` is sent instead of `null` when the schema data is `undefined`

### Type of change

If the pull request is a **bug-fix**, **enhancement** or a **refactor**, please fill in the details on the changes made.

- Existing capabilities/behavior
   If any schema field's value is undefined, null was sent as part of `payload.data` for that field.

- New capabilities/behavior
  If any schema field's value is undefined, empty string will be sent as part of `payload.data` for that field. 

If the pull request is a **new feature**,

### Any technical or performance related pointers to consider with the change?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### If the PR has changes in more than 10 files, please mention why the changes were not split into multiple PRs.

N/A

### If multiple linear tasks are associated with the PR changes, please elaborate on the reason:

N/A

<hr>

### Developer checklist

- [ ] **No breaking changes are being introduced.**

- [ ] Are all related docs linked with the PR?

- [ ] Are all changes manually tested?

- [ ] Does this change require any documentation changes?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

File: src/v0/destinations/fb_custom_audience/util.js [hunk 1]
```diff
@@ -131,21 +131,21 @@ const getUpdatedDataElement = (dataElement, isHashRequired, eachProperty, update
   /**
    * hash the original value for the properties apart from 'MADID' && 'EXTERN_ID as hashing is not required for them
    * ref: https://developers.facebook.com/docs/marketing-api/audiences/guides/custom-audiences#hash
-   * sending null values for the properties for which user hasn't provided any value
+   * sending empty string for the properties for which user hasn't provided any value
    */
   if (isHashRequired && eachProperty !== 'MADID' && eachProperty !== 'EXTERN_ID') {
```
Comment (koladilip) [inline src/v0/destinations/fb_custom_audience/util.js:136]:
```
const getUpdatedDataElement = (dataElement, isHashRequired, eachProperty, updatedProperty) => {
  let tmpUpdatedProperty = updatedProperty;
  if (eachProperty !== 'MADID' && eachProperty !== 'EXTERN_ID') {
    return dataElement;
  }
  if (!tmpUpdatedProperty) {
    dataElement.push('');
    return dataElement;
  }
  /**
   * hash the original value for the properties apart from 'MADID' && 'EXTERN_ID as hashing is not required for them
   * ref: https://developers.facebook.com/docs/marketing-api/audiences/guides/custom-audiences#hash
   * sending empty string for the properties for which user hasn't provided any value
   */
  if (isHashRequired) {
    tmpUpdatedProperty = `${tmpUpdatedProperty}`;
    dataElement.push(sha256(tmpUpdatedProperty));
  }
  return dataElement;
};
```
how about this?

Comment (sandeepdsvs) [reply src/v0/destinations/fb_custom_audience/util.js:136]: Above block doesn't look correct

Comment (sandeepdsvs) [reply src/v0/destinations/fb_custom_audience/util.js:136]: I think we can change to this
```
const getUpdatedDataElement = (dataElement, isHashRequired, eachProperty, updatedProperty) => {
  let tmpUpdatedProperty = updatedProperty;
  if (!tmpUpdatedProperty) {
    dataElement.push('');
  }
  if (eachProperty === 'MADID' || eachProperty === 'EXTERN_ID') {
    return dataElement;
  }
  /**
   * hash the original value for the properties apart from 'MADID' && 'EXTERN_ID as hashing is not required for them
   * ref: https://developers.facebook.com/docs/marketing-api/audiences/guides/custom-audiences#hash
   * sending empty string for the properties for which user hasn't provided any value
   */
  if (isHashRequired && tmpUpdatedProperty) {
    tmpUpdatedProperty = `${tmpUpdatedProperty}`;
    dataElement.push(sha256(tmpUpdatedProperty));
  }
  return dataElement;
};
```

Comment (koladilip) [reply src/v0/destinations/fb_custom_audience/util.js:136]: Yes make sense

**Pull request closed without merging**